File: app/src/main/java/ru/rsppv/criminalintent/model/Crime.java
package ru.rsppv.criminalintent.model;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.UUID;


public class Crime {
    private static final String DATE_PATTERN = "EEE, d MMM yyyy";
    private static final String TIME_PATTERN = "HH:mm";

    private UUID mId;
    private String mTitle;
    private Date mDate;
    private boolean mSolved;

    public Crime() {
        mId = UUID.randomUUID();
        mDate = new Date();
    }

    public UUID getId() {
        return mId;
    }

    public String getTitle() {
        return mTitle;
    }

    public void setTitle(String title) {
        mTitle = title;
    }

    public Date getDate() {
        return mDate;
    }

    public String getDateString() {
        SimpleDateFormat format = new SimpleDateFormat(DATE_PATTERN);
        return format.format(mDate);
    }

    public String getTimeString() {
        SimpleDateFormat format = new SimpleDateFormat(TIME_PATTERN);
        return format.format(mDate);
    }

    public void setDate(Date date) {
        mDate = date;
    }

    public boolean isSolved() {
        return mSolved;
    }

    public void setSolved(boolean solved) {
        mSolved = solved;
    }
}

File: app/src/main/java/ru/rsppv/criminalintent/model/Crime.kt
package ru.rsppv.criminalintent.model

import android.content.Context
import android.text.format.DateFormat
import java.util.*


class Crime(val id: UUID) {

    constructor() : this(UUID.randomUUID())

    var title: String? = null
    var date: Date = Date()
    var isSolved: Boolean = false
    var suspect: String? = null

    fun getDateString(context: Context?): String? {
        return DateFormat.getDateFormat(context).format(date)
    }

    fun getPhotoFilename() = "IMG_$id.jpg"
}
